WooCommerce/WooCommerceTests/ViewRelated/Orders/Payment Methods/PaymentMethodsViewModelTests.swift

```diff
@@ -135,18 +141,60 @@ final class PaymentMethodsViewModelTests: XCTestCase {
         await viewModel.markOrderAsPaidByCash(with: nil)
     }
 
+    func test_mark_order_as_paid_by_cash_then_order_status_and_payment_method_fields_updated() async {
+        // Given
+        let stores = MockStoresManager(sessionManager: .testingInstance)
+        let storage = MockStorageManager()
+        let siteID: Int64 = 10
+        let orderID: Int64 = 123
+        let order = Order.fake().copy(siteID: siteID, orderID: orderID)
+        storage.insertSampleOrder(readOnlyOrder: order)
+        let paymentGateway = PaymentGateway.defaultPayInPersonGateway(siteID: siteID).copy(title: "Pay in Person")
+        storage.insertSamplePaymentGateway(readOnlyGateway: paymentGateway)
+        let dependencies = Dependencies(stores: stores, storage: storage)
```

Comment:
I thought we wouldn't need to inject an instance of storage into the gateway dependencies for some cases, but that doesn't seem to be the case, if we do not specifically call `storage.insertSampleOrder(readOnlyOrder: .fake())` we'll see crashes in tests.

My understanding is this comes from the change from using `auto-draft` orders to `pending` orders, where auto-drafts are temporary and not meant to be stored, while pending orders are actual orders already started to be processed and should be stored. Does that sounds right?

If that's the case I think makes total sense for cash orders, but we should keep an eye to see if we've changed behaviour unintendedly where auto-drafts are used 👍

Reply:
We need to insert a sample order because we take an order from storage to pass to mody and then pass to `updateOrder` action. When we were using `updateOrderStatus`, we only needed orderID, so no order had to exist in the storage.